src/Native/System.Native/pal_runtimeinformation.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_runtimeinformation.h"
+#include "pal_types.h"
+#include <stdio.h>
+#include <sys/utsname.h>
+
+extern "C" const char* GetUnixVersion()
+{
+    struct utsname _utsname;
+    size_t length = _UTSNAME_LENGTH * 3 + 2;
+    char * version = new char[length];
```

Comment:
Where does this get freed? We should pass in a buffer from managed code with a single call to get the appropriate length once per process.